flink-streaming-scala/src/main/scala/org/apache/flink/streaming/api/scala/StreamExecutionEnvironment.scala

```diff
@@ -124,7 +125,7 @@ class StreamExecutionEnvironment(javaEnv: JavaEnv) {
    * Gets the checkpoint config, which defines values like checkpoint interval, delay between
    * checkpoints, etc.
    */
-  def getCheckpointConfig = javaEnv.getCheckpointConfig()
```

Comment:
That is fine, IDEA may have different recommendations, Flink can define its own guidelines.
I (and others) don't agree with everything that the Scala folks recommend.
And even in that description, it is phrased as a possibility to bridge the gap between Scala's and Java's features.